lib/lte_link_control/lte_lc.c

```diff
@@ -57,18 +60,26 @@ enum feaconf_feat {
 
 /* Static variables */
 
+/* Previously received LTE mode as indicated by the modem */
+static enum lte_lc_lte_mode prev_lte_mode = LTE_LC_LTE_MODE_NONE;
 /* Request eDRX to be disabled */
 static const char edrx_disable[] = "AT+CEDRXS=3";
-/* Default eDRX setting */
-static char edrx_param_ltem[5] = CONFIG_LTE_EDRX_REQ_VALUE_LTE_M;
-static char edrx_param_nbiot[5] = CONFIG_LTE_EDRX_REQ_VALUE_NBIOT;
-/* Default PTW setting */
-static char ptw_param_ltem[5] = CONFIG_LTE_PTW_VALUE_LTE_M;
-static char ptw_param_nbiot[5] = CONFIG_LTE_PTW_VALUE_NBIOT;
-/* Default PSM RAT setting */
-static char psm_param_rat[9] = CONFIG_LTE_PSM_REQ_RAT;
-/* Default PSM RPTAU setting */
-static char psm_param_rptau[9] = CONFIG_LTE_PSM_REQ_RPTAU;
+/* Requested eDRX setting */
+static char requested_edrx_value_ltem[LTE_LC_EDRX_VALUE_LEN] = CONFIG_LTE_EDRX_REQ_VALUE_LTE_M;
+static char requested_edrx_value_nbiot[LTE_LC_EDRX_VALUE_LEN] = CONFIG_LTE_EDRX_REQ_VALUE_NBIOT;
+/* Requested PTW setting */
+static char requested_ptw_value_ltem[LTE_LC_EDRX_VALUE_LEN] = CONFIG_LTE_PTW_VALUE_LTE_M;
+static char requested_ptw_value_nbiot[LTE_LC_EDRX_VALUE_LEN] = CONFIG_LTE_PTW_VALUE_NBIOT;
+/* Currently set eDRX setting as indicated by the modem */
```

Comment:
I would make it more explicit that this is the setting currently used. One could understand this so, that this is the value set using `+CEDRXS`, which is not necessarily the same as the value granted by the network. It might be enough to change the word `set` to `used`.